File: pws/database/src/main/java/com/alelk/pws/database/data/entity/BookEntity.java
package com.alelk.pws.database.data.entity;

import android.database.Cursor;

/**
 * Created by Alex Elkin on 23.04.2015.
 */
public class BookEntity implements PwsDatabaseEntity {
    private long id;
    private String name;
    private String shortName;
    private String displayName;
    private String description;
    private String version;
    private String edition;
    private String releaseDate;
    private String authors;
    private String creators;
    private String reviewers;
    private String editors;
    private String comment;
    private Integer preference;

    public BookEntity() {}

    @Override
    public long getId() {
        return id;
    }

    @Override
    public void setId(long id) {
        this.id = id;
    }

    @Override
    public PwsDatabaseEntity applyDataFomCursor(Cursor cursor) {
        // todo
        return null;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getShortName() {
        return shortName;
    }

    public void setShortName(String shortName) {
        this.shortName = shortName;
    }

    public String getDisplayName() {
        return displayName;
    }

    public void setDisplayName(String displayName) {
        this.displayName = displayName;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getVersion() {
        return version;
    }

    public void setVersion(String version) {
        this.version = version;
    }

    public String getEdition() {
        return edition;
    }

    public void setEdition(String edition) {
        this.edition = edition;
    }

    public String getReleaseDate() {
        return releaseDate;
    }

    public void setReleaseDate(String releaseDate) {
        this.releaseDate = releaseDate;
    }

    public String getAuthors() {
        return authors;
    }

    public void setAuthors(String authors) {
        this.authors = authors;
    }

    public String getCreators() {
        return creators;
    }

    public void setCreators(String creators) {
        this.creators = creators;
    }

    public String getReviewers() {
        return reviewers;
    }

    public void setReviewers(String reviewers) {
        this.reviewers = reviewers;
    }

    public String getEditors() {
        return editors;
    }

    public void setEditors(String editors) {
        this.editors = editors;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    public Integer getPreference() {
        return preference;
    }

    public void setPreference(Integer preference) {
        this.preference = preference;
    }

    @Override
    public String toString() {
        return "BookEntity{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", shortName='" + shortName + '\'' +
                ", displayName='" + displayName + '\'' +
                ", description='" + description + '\'' +
                ", version='" + version + '\'' +
                ", edition='" + edition + '\'' +
                ", releaseDate='" + releaseDate + '\'' +
                ", authors='" + authors + '\'' +
                ", creators='" + creators + '\'' +
                ", reviewers='" + reviewers + '\'' +
                ", editors='" + editors + '\'' +
                ", comment='" + comment + '\'' +
                ", preference='" + preference + '\'' +
                '}';
    }
}

File: database/src/commonMain/kotlin/io/github/alelk/pws/database/common/entity/BookEntity.kt
package io.github.alelk.pws.database.common.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import io.github.alelk.pws.database.common.model.BookExternalId
import io.github.alelk.pws.database.common.model.Person
import io.github.alelk.pws.database.common.model.Version
import io.github.alelk.pws.database.common.model.Year
import java.util.Locale

@Entity(
  tableName = "books",
  indices = [Index(name = "idx_books_edition", value = ["edition"], unique = true)]
)
data class BookEntity(
  @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "_id") val id: Long? = null,
  @ColumnInfo(name = "version") val version: Version,
  @ColumnInfo(name = "locale") val locale: Locale,
  @ColumnInfo(name = "name") val name: String,
  @ColumnInfo(name = "displayshortname") val displayShortName: String,
  @ColumnInfo(name = "displayname") val displayName: String,
  @ColumnInfo(name = "edition") val externalId: BookExternalId,
  @ColumnInfo(name = "releasedate") val releaseDate: Year? = null,
  @ColumnInfo(name = "authors") val authors: List<Person>? = null,
  @ColumnInfo(name = "creators") val creators: List<Person>? = null,
  @ColumnInfo(name = "reviewers") val reviewers: List<Person>? = null,
  @ColumnInfo(name = "editors") val editors: List<Person>? = null,
  @ColumnInfo(name = "description") val description: String? = null,
  @ColumnInfo(name = "preface") val preface: String? = null
)